File: ios/Utils/Mapper/ImpressionDataMapper.swift
import Foundation
import YandexMobileAds

private let rawDataKey = "rawData"

class ImpressionDataMapper {
    func map(_ impressionData: ImpressionData) -> [String: Any] {
        let result: [String: Any] = [
            rawDataKey: impressionData.rawData
        ]
        return result
    }
}

File: android/src/main/java/com/yandexmobileads/util/mapper/ImpressionDataMapper.kt
package com.yandexmobileads.util.mapper

import com.yandex.mobile.ads.common.ImpressionData

internal class ImpressionDataMapper {

    fun map(impressionData: ImpressionData): Map<String, Any> = mapOf(
        KEY_RAW_DATA to impressionData.rawData
    )

    companion object {
        private const val KEY_RAW_DATA = "rawData"
    }
}
